### PR TITLE
Problem: (fix #64)Missing docs for nitro-enclave tmkms

### LINKED ISSUE #64: Problem: Missing docs for nitro-enclave tmkms
it should include how to set up nitro-enclave environment
run tmkms on nitro-enclave

### PR DESCRIPTION
**Pull request by allthatjazzleo**
Solution: add the setup and configuration procedure.

For Step 5.4, as it is not yet merged to master.
you may need to modify the url or directly use the script without `curl`

### CODE REVIEW

**Review by tomtau — CHANGES_REQUESTED** [review]
I'd probably add some warning that this isn't yet ready for production use: 1) not yet audited, 2) that tmkms prototype fork isn't meant to be maintained in the long term

File: docs/getting-started/notes-on-production-deployment.md [hunk 1]
```diff
@@ -1,51 +1,270 @@
...
+```
+Take a note of the **encrypted validator signing key** from `ciphertext`
+
+#### Step 5.4. Obtain the consensus / validator public key
+If you plan to run the tmkms enclave without debug mode, you will not able to access to the tmkms log. Normally, you shall obtain the `consensus / validator public key` from the tmkms log. Therefore, before running the tmkms enclave, you shall obtain the consensus / validator public key in advance from the local or trusted machine where you also keep the generated `signing.key`.
+
+```bash
+curl https://raw.githubusercontent.com/crypto-com/chain-docs/master/docs/resources/obtain-tmkms-consensus-key.sh -s | bash
+```
+Take a note of the **consensus / validator public key** `tcrocnclconspub....` for joining validator later.
```
Comment (lezzokafka) [inline docs/getting-started/notes-on-production-deployment.md:121]:
```suggestion
Take a note of the **consensus / validator public key** with the prefix `tcrocnclconspub....` for joining the network as a validator later.
```

File: docs/getting-started/notes-on-production-deployment.md [hunk 2]
```diff
@@ -1,51 +1,270 @@
...
 
-1. strip off the Amino prefix; in the case of Ed25519 public keys, it's 5 bytes: 0x16, 0x24, 0xDE, 0x64, 0x20.
+### Step 9. Running chain-maind
 
-2. convert the stripped-off `KEY_HEX_PAYLOAD` to base64.
+One should follow the same steps in [Croeseid Testnet: Running Nodes](./croeseid-testnet.md)
 
-#### Step 2. place / use the public key where needed
-For example, generate the `genesis.json` with it if the corresponding node is one of the initial validators / council nodes.
+Except one last thing one need to further configure `~/.chain-maind/config/config.toml` to enable enclave tmkm to sign.
```
Comment (lezzokafka) [inline docs/getting-started/notes-on-production-deployment.md:268]:
```suggestion
Except for one last thing one needs to further configure `~/.chain-maind/config/config.toml` to enable enclave tmkm to sign.
```

**Review by lezzokafka — COMMENTED** [review]
Looks great, some comment first:

**Comment by allthatjazzleo** [comment]
fixed @tomtau @lezzokafka

**Review by lezzokafka — APPROVED** [review]
Thanks for the great effort!

File: docs/getting-started/advanced-tmkms-integration.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+::: warning CAUTION
+The setup isn't yet ready for production use: 
+1) It is not yet audited
+2) The [tmkms prototype fork](https://github.com/tomtau/tmkms/tree/feature/nitro-enclave) isn't meant to be maintained in the long term
+:::
+
+### Background
+[TMKMS](https://github.com/iqlusioninc/tmkms), initially targeting Cosmos Validators, provide **High-availability**, **Double-signing prevention** and **Hardware security module**.
+
+Currently, TMKMS provides both **hardware signing** and **softsign**.
```
Comment (tomtau) [inline docs/getting-started/advanced-tmkms-integration.md:16]:
```suggestion
Currently, TMKMS provides both **hardware signing** and **softsign**.
 However, it is hard or impossible to plug your own [Hardware Security Modules(HSM)](https://github.com/iqlusioninc/tmkms#hardware-security-modules-recommended) to the major cloud providers when one wants to run it on the cloud for **hardware signing**. On the other hand, it is also insecure to use **softsign** as your generated signing key is actually in plain text on the machine.
```

File: docs/getting-started/advanced-tmkms-integration.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+However, it is hard or impossible to plug your own [Hardware Security Modules(HSM)](https://github.com/iqlusioninc/tmkms#hardware-security-modules-recommended) to the major cloud providers when one want to run it on the cloud for **hardware signing**. On the other hand, it is also insecure to use **softsign** as your generated signing key is actually in plain text on the machine.
```
Comment (tomtau) [inline docs/getting-started/advanced-tmkms-integration.md:17]:
```suggestion
Currently, TMKMS provides both **hardware signing** and **softsign**.
 However, it is hard or impossible to plug your own [Hardware Security Modules(HSM)](https://github.com/iqlusioninc/tmkms#hardware-security-modules-recommended) to the major cloud providers when one wants to run it on the cloud for **hardware signing**. On the other hand, it is also insecure to use **softsign** as your generated signing key is actually in plain text on the machine.
```

**Pull request merged**